wgsl/index.bs

```diff
@@ -4180,6 +4180,242 @@ TODO: deduplicate these tables
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`texture`<td>
+  The [sampled](#sampled-texture-type) or [depth](#texture-depth) texture to
+  sample.
+  <tr><td>`sampler`<td>
+  The [sampler type](#sampler-type).
+  <tr><td>`coords`<td>
+  The texture coordinates used for sampling.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index to sample.
+  <tr><td>`ddx`<td>
+  The x direction derivative vector used to compute the sampling locations.
+  <tr><td>`ddy`<td>
+  The y direction derivative vector used to compute the sampling locations.
+  <tr><td>`offset`<td>
+  The optional texel offset applied to the unnormalized texture coordinate
+  before sampling the texture. This offset is applied before applying any
+  texture wrapping modes. `offset` values may range from `[-8..7]` in all
+  dimensions. Values outside of this range will be clamped to the nearest limit.
+</table>
+
+**Returns:**
+
+The sampled value.
+
+
+### `textureSampleCompare` ### {#texturesamplecompare}
+
+Samples a depth texture and compares the sampled depth values against a reference value.
+
+```rust
+textureSampleCompare(texture : texture_depth_2d,         sampler : sampler_comparison, coords : vec2<f32>,                    depth_ref : f32                    ) -> f32
+textureSampleCompare(texture : texture_depth_2d,         sampler : sampler_comparison, coords : vec2<f32>,                    depth_ref : f32, offset : vec2<i32>) -> f32
+textureSampleCompare(texture : texture_depth_2d_array,   sampler : sampler_comparison, coords : vec2<f32>, array_index : u32, depth_ref : f32                    ) -> f32
+textureSampleCompare(texture : texture_depth_2d_array,   sampler : sampler_comparison, coords : vec2<f32>, array_index : u32, depth_ref : f32, offset : vec2<i32>) -> f32
+textureSampleCompare(texture : texture_depth_cube,       sampler : sampler_comparison, coords : vec3<f32>,                    depth_ref : f32                    ) -> f32
+textureSampleCompare(texture : texture_depth_cube_array, sampler : sampler_comparison, coords : vec3<f32>, array_index : u32, depth_ref : f32                    ) -> f32
+```
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`texture`<td>
+  The [depth](#texture-depth) texture to sample.
+  <tr><td>`sampler`<td>
+  The [sampler comparision](#sampler-type) type.
+  <tr><td>`coords`<td>
+  The texture coordinates used for sampling.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index to sample.
+  <tr><td>`depth_ref`<td>
+  The reference value to compare the sampled depth value against.
+  <tr><td>`offset`<td>
+  The optional texel offset applied to the unnormalized texture coordinate
+  before sampling the texture. This offset is applied before applying any
+  texture wrapping modes. `offset` values may range from `[-8..7]` in all
+  dimensions. Values outside of this range will be clamped to the nearest limit.
+</table>
+
+**Returns:**
+
+A `f32` in the range `[0.0..1.0]`, representing the number of depth values
```

Comment:
reads weird that a number less than 1.0 would represent "number of depth values". Maybe we can reword this better?

Reply:
Any better?